src/Swarm/Game/Step.hs

```diff
@@ -864,10 +864,13 @@ execConst c vs k = do
             , commaList (map (^. entityName) (S.toList missingDevices))
             ]
 
-        -- update other robot's CEK machine and environment
-        lift . lift $ robotMap . at otherRobotName . _Just . machine .= In cmd e [FExec]
-        lift . lift $ robotMap . at otherRobotName . _Just . robotEnv .= empty
-        lift . lift $ robotMap . at otherRobotName . _Just . robotCtx .= (empty, empty)
+        -- update other robot's CEK machine, environment and context
+        -- the childRobot inherits the parent robot's environment
+        -- and context which collectively mean all the variables
+        -- declared in the parent robot
+        lift . lift $ robotMap . at childRobotName . _Just . machine .= In cmd e [FExec]
+        lift . lift $ robotMap . at childRobotName . _Just . robotEnv .= e
```

Comment:
This should probably not be `e`, but rather should be literally whatever is stored in the `robotEnv` of the parent robot.  The difference is that the current environment `e` might have additional local bindings beyond what is in the `robotEnv`.  Like if the base has defined a command `c` but then this `reprogram` command was run like `let x = 3 in reprogram ... ` Now the environment `e` will contain both `c` and `x`.  However in this case the definition of `x` is supposed to be local to this computation, we wouldn't want to keep it around for later programs run on the same robot.

As discussed elsewhere, it ultimately doesn't really matter right now, but we might as well do "the most sensible thing" in case it turns out to matter with changes we make later!